### DISCUSSION TITLE
How to send User Profile Name while using magic link

### DISCUSSION THREAD

**Discussion by surjithctly** [nextauthjs/next-auth#5244] [Help]
I'm using `magic-link` Email Auth with Dynamodb. I wanted to send User's First Name & custom ID to the database. How do I do that?

As per the docs: it should be populated: 
https://next-auth.js.org/configuration/callbacks#sign-in-callback
<img width="997" alt="CleanShot 2022-08-29 at 15 33 27@2x" src="https://user-images.githubusercontent.com/1884712/187177130-0d7c009e-5e20-4fc5-a000-506c3bb565d3.png">

Here's what I have tried: 

```js
// not working
 const { error } = await signIn("email", {
        email,
        id: "123",
        name: "John", 
        callbackUrl: `${window.location.origin}/dashboard`,
      });

// not working
 const { error } = await signIn("email", {
        email,
        user: {
          id: "123",
          name: "John", 
        }
        callbackUrl: `${window.location.origin}/dashboard`,
      });

// not working
 const { error } = await signIn("email", {
        email,
        profile: {
          id: "123",
          name: "John", 
        }
        callbackUrl: `${window.location.origin}/dashboard`,
      });
```

**Comment by Abel1011** [comment]
Hello, you can pass the additional parameters as indicated here: https://next-auth.js.org/getting-started/client#additional-parameters. Then through advanced initialization (see https://next-auth.js.org/configuration/initialization#advanced-initialization) you can access those parameters through the request body.

**Reply by surjithctly** [reply]
If you see my code, I have used the additional params. But its not working as expected.

**Reply by Abel1011** [reply]
You should send the parameters as indicated in the documentation, for your use case it would be:

```
 const { error } = await signIn("email", {
        email,
        callbackUrl: `${window.location.origin}/dashboard`,
      },{
        id: "123",
        name: "John", 
});
```

You will find those parameters in the req.body or in the req.query, I don't remember exactly where they are passed.
